src/pages/profiles/CreateProfile.tsx

```diff
@@ -299,6 +303,76 @@ const CreateProfile: FC = () => {
           Create
         </ActionButton>
       </FormFooterLayout>
+
+      {panelParams.panel === panels.createNetworkDevice && (
+        <CreateNetworkDevicePanel
+          project={project}
+          formik={formik}
+          onClose={panelParams.clear}
+          onSave={() => {
+            const deviceIndex = formik.values.devices.length - 1;
+            const focusField = `devices.${deviceIndex}.name`;
+            setTimeout(() => {
+              const element = document.querySelector(
+                `[name="${focusField}"]`,
+              ) as HTMLInputElement;
+              if (element) {
+                element.focus();
+              }
+            }, 100);
+          }}
+        />
+      )}
+
+      {panelParams.panel === panels.editNetworkDevice &&
+        panelParams.deviceName &&
+        (() => {
+          const urlParams = new URLSearchParams(window.location.search);
+          const createOverride = urlParams.get("create-override") === "true";
+
+          const deviceIndex = formik.values.devices.findIndex(
+            (d) => d.name === panelParams.deviceName,
+          );
+          const device = formik.values.devices[deviceIndex];
+
+          // Handle create override mode for inherited networks
+          if (createOverride && deviceIndex === -1) {
+            const tempDevice = {
+              name: panelParams.deviceName,
+              type: "nic",
+              network: "",
+            };
+
+            return (
+              <EditNetworkDevicePanel
+                device={tempDevice as FormNetworkDevice}
+                deviceIndex={-1}
+                project={project}
+                formik={formik}
+                onClose={panelParams.clear}
+                onSave={(updatedDevice) => {
+                  const newDeviceIndex = formik.values.devices.length;
+                  formik.setFieldValue(
+                    `devices.${newDeviceIndex}`,
+                    updatedDevice,
+                  );
+                  panelParams.clear();
+                }}
+              />
+            );
+          }
+
+          return deviceIndex !== -1 && device ? (
+            <EditNetworkDevicePanel
+              device={device as FormNetworkDevice}
+              deviceIndex={deviceIndex}
+              project={project}
+              formik={formik}
+              onClose={panelParams.clear}
+              onSave={panelParams.clear}
+            />
+          ) : null;
+        })()}
```

Comment:
This should be using the same component between edit/create instance and profile, not duplicating the logic for all four cases.